File: backend/routes/media.py
from fastapi import APIRouter, UploadFile, File, Form

from services.media_service import upload_media

router = APIRouter(
    prefix="/media",
    tags=["Media"]
)


@router.post("/upload")
async def upload_file(
    file: UploadFile = File(...),
    folder: str = Form(...)
):
    result = await upload_media(
        file=file.file,
        folder=folder
    )

    return {
        "success": True,
        "data": result
    }

File: frontend-shell/src/api/media.js
import api from "./client";

export const uploadMedia = async (
    file,
    folder = "uploads"
) => {
    const formData = new FormData();

    formData.append("file", file);
    formData.append("folder", folder);

    const res = await api.post(
        "/media/upload",
        formData,
        {
            headers: {
                "Content-Type":
                    "multipart/form-data",
            },
        }
    );

    return res.data;
};